File: daily-coding-problems/1/main.ts
function sumOfTwoBrute(list: number[], k: number): boolean
{
    for(let i = 0; i < list.length; i++) {
        const n = JSON.parse(JSON.stringify(list));
        n.splice(i, 1);

        const value = list[i];
        for(let j = 0; j < n.length; j++) {
            if((value + n[j]) == k) {
                return true;
            }
        }
    }

    return false;
}

function sumOfTwo(list: number[], k: number): boolean
{
    const r = {};

    for(let i = 0; i < list.length; i++) {
        let c = list[i],
            sub = k - c;

        if(r[sub])
            return true;

        r[c] = true;
    }

    return false;
}


const L = [10, 15, 3, 7];
const K = 18;

console.log(sumOfTwo(L, K));

File: daily-coding-problems/1/main.go
package main

import "fmt"

func sumOfTwoBrute(list []int, k int) bool {

    for i, va := range list {
        n := list
        n = append(n[:i], n[i + 1:]...)

        for _, vb := range n {
            if (va + vb) == k {
                return true
            }
        }
    }

    return false;
}


func sumOfTwo(list []int, k int) bool {

    r := make(map[int]bool)

    for _, v := range list {

        sub := k - v
        if r[sub] {
            return true
        }

        r[v] = true
    }

    return false
}


func main() {

    list := []int{10, 15, 3, 7}
    k := 13

    fmt.Println(sumOfTwo(list, k))
}
